src/Illuminate/Filesystem/FilesystemAdapter.php

```diff
@@ -847,6 +847,10 @@ public function temporaryUploadUrl($path, $expiration, array $options = [])
      */
     protected function concatPathToUrl($url, $path)
     {
+        if (empty($url) || empty($path)) {
```

Comment:
Empty is the wrong function here. It does not check if the variable us the empty string, it checks if it looks like an enmpty-ish value, inc if it's 0. One should compare against `''` instead.